### PR TITLE
Add support for MSTest to CookieCrumble / Snapshot

### PR DESCRIPTION
**Pull request by stuartjbrown**
Adds support for MSTest testing alongside XUnit for Snapshot.  This works by:

- The test info (assembly name and test type) is determined in the Snapshot constructor
- This info is used later when raising exceptions of the relevant type for the test

### CODE REVIEW

**Comment by michaelstaib** [comment]
Thank you for contributing!

File: src/CookieCrumble/src/CookieCrumble/CookieCrumble.csproj [hunk 1]
```diff
@@ -15,6 +15,7 @@
     <PackageReference Include="Microsoft.AspNetCore.WebUtilities" Version="2.2.0" />
     <PackageReference Include="Xunit" Version="2.4.1" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+    <PackageReference Include="MSTest.TestFramework" Version="3.0.4" />
```
Comment (michaelstaib) [inline src/CookieCrumble/src/CookieCrumble/CookieCrumble.csproj:18]: I do not want to have mstest referenced in all of our tests ... we probably should abstract this more and move things around a bit. I will think on this a bit.

Comment (stuartjbrown) [reply src/CookieCrumble/src/CookieCrumble/CookieCrumble.csproj:18]: I did think about splitting it out into separate assemblies for each test framework - so one shared assembly with most of the logic, then one for XUnit, one for MSTest etc (a bit like how the Snapshot test assembly was previously)

Comment (stuartjbrown) [reply src/CookieCrumble/src/CookieCrumble/CookieCrumble.csproj:18]: Refactored the code in a follow-up PR (https://github.com/ChilliCream/graphql-platform/pull/6537) so we can close this one

**Comment by glen-84** [comment]
Superseded by #6537.

**Pull request closed without merging**